src/settingsholder.cpp

```diff
@@ -305,3 +316,24 @@ void SettingsHolder::addVpnDisabledApp(const QString& appID) {
   applist.append(appID);
   setVpnDisabledApps(applist);
 }
+
+bool SettingsHolder::isValidUserDNS(const QString& dns) {
+  logger.log() << "checking -> " << dns;
+  QHostAddress address = QHostAddress(dns);
+
+  logger.log() << "is null " << address.isNull();
+
+  if (address.isNull()) {
+    return false;
+  }
+  /* Currently we need to limit this to LAN-DNS
+   * (at least on windows) since the killswitch makes
+   * sure that no dns traffic may happen to outside of lan
+   */
```

Comment:
Can you tell me more about this? Why cannot we have an external DNS?

Reply:
The problem here seems to be the WFP rules, wg-go is setting. So Even if we exclude <some-ip>, dns traffic to it is blocked. If wanted we can easily override it.